otelcolbuilder/Makefile

```diff
@@ -1,5 +1,5 @@
 BINARY_NAME ?= otelcol-sumo
-BUILDER_VERSION ?= 0.102.1
+BUILDER_VERSION ?= 0.103.1
```

Comment:
Shouldn't be `0.103.0`? 🤔

Reply:
otel collector Builder 0.103.0 is not released, only 0.103.1 is available